### ISSUE TITLE
login middleware authenticates all clients

### ISSUE THREAD

**Issue by sk91** [elpheria/rpc-websockets#64]
Hi,
Trying to understand how login middlewere should work.

Came across this code:

```
// if login middleware returned true, set connection as authenticated
        if (message.method === "rpc.login" && response === true)
            this.authenticated = true
```

Does it mean that  after one client authenticates, all clients are authenticated automatically?

**Comment by mkozjak** [comment]
No, only that specific client is authenticated.

**Comment by sk91** [comment]
Sorry, @mkozjak it does not seems so. What am I missing?

here is a test code:
```
var WebSocket = require('rpc-websockets').Client
var WebSocketServer = require('rpc-websockets').Server

const initServer = ()=>{
    var server = new WebSocketServer({
        port: 8080,
        host: 'localhost'
      })
      
      // register an RPC method
      server.register('sum', function(params) {
        console.log('XXX', 'public method call', 'sum', params)
        return params[0] + params[1]
      })
      
      // ...and maybe a protected one also
      server.register('account', function() {
        console.log('XXX', 'protected method call', 'account')
        return ['confi1', 'confi2']
      }).protected()

      server.setAuth(()=>true);
      

      return server;
}

const initClient = async ()=>{
    var ws = new WebSocket('ws://localhost:8080');
    return new Promise((resolve, reject)=>{
        ws.on('open', ()=>{
            resolve(ws);
        });
        ws.on('error', error=>reject(error));
    });
}

const main = async ()=>{
    try {
    const server = initServer();
    console.log('XXX', 'server initialized');

    const ws1 = await initClient();
    console.log('XXX', 'client1 initialized');

    const ws2 = await initClient();
    console.log('XXX', 'client2 initialized');

    try {
        const result =  await ws1.call('account');
        console.log('XXX', 'client 1', 'call for account before login succesfull', result);
    } catch(error){
        console.log('XXX', 'client 1', 'call for account before login failed', error);
    }

    try {
        const result =  await ws2.call('account');
        console.log('XXX', 'client 2', 'call for account before login succesfull', result);
    } catch(error){
        console.log('XXX', 'client 2', 'call for account before login failed', error);
    }

    await ws1.login({'username': 'confi1', 'password':'foobar'});


    try {
        const result =  await ws1.call('account');
        console.log('XXX', 'client 1', 'call for account after client 1 login succesfull', result);
    } catch(error){
        console.log('XXX', 'client 1', 'call for account after client 1 login failed', error);
    }

    try {
        const result =  await ws2.call('account');
        console.log('XXX', 'client 2', 'call for account after client 1 login succesfull', result);
    } catch(error){
        console.log('XXX', 'client 2', 'call for account after client 1 login failed', error);
    }
    
    }
    catch(error){
        console.log('global error occurred', error);
    }
}

main();
```

and here is the output:

```
XX server initialized
XXX client1 initialized
XXX client2 initialized
XXX client 1 call for account before login failed { code: -32605, message: 'Method forbidden' }
XXX client 2 call for account before login failed { code: -32605, message: 'Method forbidden' }
XXX protected method call account
XXX client 1 call for account after client 1 login succesfull [ 'confi1', 'confi2' ]
XXX protected method call account
XXX client 2 call for account after client 1 login succesfull [ 'confi1', 'confi2' ]
```

**Comment by mkozjak** [comment]
@sk91: You're definitely correct, good catch!

We store clients in `this.namespaces[ns].clients` so maybe we could find some place to store the authenticated info there? Would you like to provide a PR to fix this bug? Thanks in advance!

**Comment by sk91** [comment]
@mkozjak, researching now a solution to create a mobile <-> desktop communication over LAN.
This library seems like a good one to go with.

If we decide on it, I will do the fix, it seems simple enough

**Comment by mkozjak** [comment]
@sk91: Okay, also feel free to provide a PR to add your company to the list of users if you see fit: https://github.com/elpheria/rpc-websockets#users

**Comment by dsagal** [comment]
This issue means that login/setAuth cannot be used at the moment. They should be removed from documentation until there is a way to use them.

**Comment by mkozjak** [comment]
@dsagal - the bugfix is available in v5.2.2.
